File: python/7 kyu/116.py
# https://www.codewars.com/kata/588425ee4e8efb583d000088/train/python

# Passed

def phone_call(min1, min2_10, min11, s):
    current_minute = 1
    while True:
        if current_minute == 1:
            s -= min1
        elif current_minute > 1 and current_minute < 11:
            s -= min2_10
        else:
            s -= min11
        
        if s <= 0:
            if s < 0:
                current_minute -= 1 # last minute isn't valid
            break        

        current_minute += 1

    return current_minute

output = phone_call(3,1,2,20)
print(output)

File: nodejs/6 kyu/116.js
// https://www.codewars.com/kata/551186edce486caa61000f5c/train/javascript

// Passed

function countSquares(n, memo = {}) {
  if (n === 1) return 1;

  if (n in memo) {
    return memo[n];
  }

  let result = countSquares(n - 1, memo) + n ** 2;
  memo[n] = result;
  return result;
}

const output = countSquares(8);
console.log(output);